test/system/shoot_cp_migration/migrate_test.go

```diff
@@ -0,0 +1,254 @@
+	if err = t.GardenerFramework.GetShoot(ctx, shoot); err != nil {
+		return err
+	}
+
+	kubecfgSecret := corev1.Secret{}
+	if err := t.GardenerFramework.GardenClient.DirectClient().Get(ctx, client.ObjectKey{Name: shoot.Name + ".kubeconfig", Namespace: shoot.Namespace}, &kubecfgSecret); err != nil {
+		t.GardenerFramework.Logger.Errorf("Unable to get kubeconfig from secret: %s", err.Error())
+		return err
+	}
+	t.GardenerFramework.Logger.Info("Shoot kubeconfig secret was fetched successfully")
+
+	t.ShootClient, err = kubernetes.NewClientFromSecret(ctx, t.GardenerFramework.GardenClient.DirectClient(), kubecfgSecret.Namespace, kubecfgSecret.Name, kubernetes.WithClientOptions(client.Options{
+		Scheme: kubernetes.ShootScheme,
+	}))
+	t.Shoot = *shoot
+	return
+}
+
+func initSeedsAndClients(ctx context.Context, t *ShootMigrationTest) error {
+	t.Config.SourceSeedName = *t.Shoot.Spec.SeedName
+
+	if seed, seedClient, err := t.GardenerFramework.GetSeed(ctx, t.Config.TargetSeedName); err != nil {
+		return err
+	} else {
+		t.TargetSeedClient = seedClient
+		t.TargetSeed = seed
+	}
+
+	if seed, seedClient, err := t.GardenerFramework.GetSeed(ctx, t.Config.SourceSeedName); err != nil {
+		return err
+	} else {
+		t.SourceSeedClient = seedClient
+		t.SourceSeed = seed
+	}
+	return nil
+}
+
+func beforeMigration(ctx context.Context, t *ShootMigrationTest, guestBookApp *applications.GuestBookTest, testSecret *corev1.Secret, testServiceAccount *corev1.ServiceAccount) error {
+	ginkgo.By(fmt.Sprintf("Initializing Shoot %s/%s and its Client", *shootNamespace, *shootName))
+	if err := initShootAndClient(ctx, t); err != nil {
+		return err
+	}
+	t.SeedShootNamespace = ComputeTechnicalID(t.GardenerFramework.ProjectNamespace, &t.Shoot)
+
+	ginkgo.By(fmt.Sprintf("Initializing source Seed %s, target Seed %s, and their Clients", *t.Shoot.Spec.SeedName, *targetSeedName))
+	if err := initSeedsAndClients(ctx, t); err != nil {
+		return err
+	}
+
+	ginkgo.By("Fetching the objects that will be used for comparison")
+	t.ComparisonElementsBeforeMigration = ShootComparisonElements{}
+	if err := t.PopulateBeforeMigrationComparisonElements(ctx); err != nil {
+		return err
+	}
+
+	ginkgo.By("Creating test Secret and Service Account")
+	if err := t.ShootClient.DirectClient().Create(ctx, testSecret); err != nil {
+		return err
+	}
+	t.GardenerFramework.Logger.Infof("Secret resource %s/%s was created!", testSecret.Name, testSecret.Namespace)
+	if err := t.ShootClient.DirectClient().Create(ctx, testServiceAccount); err != nil {
+		return err
+	}
+	t.GardenerFramework.Logger.Infof("ServiceAccount resource %s/%s was created!", testServiceAccount.Name, testServiceAccount.Namespace)
+
+	ginkgo.By("Deploying Guest Book Application")
+	initializedApp, err := initGuestBookTest(ctx, t)
+	if err != nil {
+		return err
+	}
+	*guestBookApp = *initializedApp
+	guestBookApp.DeployGuestBookApp(ctx)
```

Comment:
I think we do not need to run the guestbook test here, as I think we run the whole gardener shoot test suite after the migration which already includes the guestbook test

Reply:
The idea here was to add a function that could continuously try to run the guest book app test during migration to be sure that there aren't any interruptions. I didn't comment on it as we could add it in a future PR.

Reply:
ah ok so you want to check whether a deployed guestbook is successfully migrated and working on a new seed.
Then we need it here

Reply:
cc @dguendisch

Reply:
Sure, didn't thought about that purpose, makes sense, so keep it in :)